src/features/dashboard/components/ApiTokenCard/index.tsx

```diff
@@ -1,41 +1,93 @@
+import { Text, Heading, Modal, SectionMessage } from '@deriv-com/quill-ui';
+import { StandaloneCircleExclamationRegularIcon } from '@deriv/quill-icons';
+import { TApiTokenForm, TApiTokenFormItemsNames } from '../ApiTokenForm/api-token.form';
 import styles from './api-token.card.module.scss';
 
-interface IApiTokenCardPros extends HTMLAttributes<HTMLDivElement> {
+interface IApiTokenCardProps {
   register: UseFormRegister<TApiTokenForm>;
   name: TApiTokenFormItemsNames;
   label: string;
   description: string;
 }
 
-const ApiTokenCard = ({ register, name, label, description, ...rest }: IApiTokenCardPros) => {
-  return (
-    <div className={clsx(styles.api_token_card)} {...rest}>
-      <div className={styles.api_token_card_input}>
-        <CustomCheckbox name={name} id={`${name}-scope`} register={register(name)}>
-          <label data-testid={`card-label-${name}`} htmlFor={`${name}-scope`}>
-            <b> {label} </b>
-          </label>
-        </CustomCheckbox>
-      </div>
-      <Text role={'definition'} as={'p'} type={'small'} className={styles.description}>
-        {description}
-      </Text>
-      <React.Fragment>
-        {name === 'admin' && (
-          <div className={styles.warning_container}>
-            <img className={styles.warning_image} src='/img/warning.svg' />
-            <p>
-              <b>Note:</b> Do not share tokens with the Admin scope with unauthorised parties.
-            </p>
+const ApiTokenCard = ({ register, name, label, description }: IApiTokenCardProps) => {
+  const [isAdminChecked, setIsAdminChecked] = useState(false);
+  const [isAdminPopupVisible, setIsAdminPopupVisible] = useState(false);
+  const { deviceType } = useDeviceType();
+
+  const handleAdminScopeChange = (e?: React.ChangeEvent<HTMLInputElement>, chk?: boolean) => {
+    if (e) {
+      const isChecked = e.target.checked;
+      setIsAdminChecked(isChecked);
+      setIsAdminPopupVisible(isChecked);
+    } else if (chk) {
+      setIsAdminPopupVisible(false);
+      setIsAdminChecked(true);
+    } else {
+      setIsAdminPopupVisible(false);
+      setIsAdminChecked(false);
+    }
+  };
+
+  const adminSection = () => {
+    return (
+      <>
+        <SectionMessage
+          message='Do not share tokens with the admin scope with unauthorized parties.'
+          size='md'
+          status='warning'
+          className='mst'
+        />
+        <Modal
+          isOpened={isAdminPopupVisible}
+          primaryButtonLabel='Enable admin access'
+          secondaryButtonLabel='Cancel'
+          primaryButtonCallback={() => handleAdminScopeChange(undefined, true)}
+          secondaryButtonCallback={() => handleAdminScopeChange()}
+          isMobile={deviceType !== 'desktop'}
+          showSecondaryButton
+          shouldCloseOnSecondaryButtonClick
+          showHandleBar
+          disableCloseOnOverlay={false}
+        >
+          <div className='adminScopePopup__icons'>
```

Comment:
can't we use global properties like `modal__icons` and `modal__content` ?

Reply:
Done for the content. The adminscopePopup__icons has the background behind the icon and popups it's different for each of them if it's a warning (yellow) success (green) or failure (red) popup.